src/main/java/com/project/Investment/App/generatingModule/GeneratorPosition.java

```diff
@@ -0,0 +1,199 @@
+package com.project.Investment.App.generatingModule;
+
+import com.project.Investment.App.model.Position;
+import com.project.Investment.App.model.embeddedId.PositionId;
+import com.project.Investment.App.util.math.BigFunctions;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import java.time.LocalDate;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.counting;
+
+@Component
+@Slf4j
+public class GeneratorPosition {
+
+    public List<Position> generatorPosition(LocalDate startDate, String entityId, int total) {
```

Comment:
start method name with a verb. in this case, you can rename it to `generatePositions`
check other method names as well

Reply:
done

Reply:
Your method still uses noun instead of verb - `generatorPositions`

Reply:
Done